### PR TITLE
feat(main-content): added mobile friendly main-content component

### PR DESCRIPTION
**Pull request by sydbails**
##Description
Mobile friendly functional main component.

##Type of change
[x]New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: src/components/MainContent/MainContent.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+          </span>
+          <Button>Sign up</Button>
+        </div>
+        <div className="wrapper--signup">
+          <h3 className="section__title--subtle">Here&#39;s how it works:</h3>
+          {/* use list provided by carbon design system. */}
+          <ol className="step__list">
+            {props.steps.map((step) => {
+              return (
+                <li className="step__list-item">
```
Comment (mmhuntsberry) [inline src/components/MainContent/MainContent.js]: Moving from desktop to large tablet breakpoint your bottom margin jumps around.  You have no bottom margin when you first break your steps into a two by two grid but once you get down to tablet size then bottom margin appears.  Should have margin once it breaks into  two by two grid.

File: src/components/MainContent/_maincontent.scss [hunk 2]
```diff
@@ -0,0 +1,132 @@
+//variables
+$gradient-background: linear-gradient(
+  90deg,
+  rgba(201, 222, 255, 1) 0%,
+  rgba(230, 214, 255, 1) 50%,
+  rgba(243, 243, 243, 1) 100%
+);
+
+.section {
+  padding: 80px 0;
```
Comment (mmhuntsberry) [inline src/components/MainContent/_maincontent.scss]: use spacing/layout token with calc if needed.

File: src/components/MainContent/_maincontent.scss [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+}
+
+.card__icon {
+  height: 64px;
+  width: 64px;
+}
+
+.step__list {
+  font-size: 1.75rem;
+  //   content: counter(item) ' ';
```
Comment (mmhuntsberry) [inline src/components/MainContent/_maincontent.scss]: remove commented out code if its unnecessary

File: src/components/MainContent/_maincontent.scss [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+  @include carbon--type-style('body-short-01');
+}
+
+.card__icon {
+  height: 64px;
+  width: 64px;
+}
+
+.step__list {
+  font-size: 1.75rem;
```
Comment (mmhuntsberry) [inline src/components/MainContent/_maincontent.scss:120]: Although, I don't agree with this our coding standards specify always using a hardcoded value for font-size

https://github.ibm.com/cloud-integration-design/coding-standards#pixels-vs-ems

File: src/components/MainContent/_maincontent.scss [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+  //   flex-direction: column;
+}
+
+.wrapper--signup {
+  display: flex;
+  flex-direction: column;
+  border-left: {
+    color: #000;
+    style: solid;
+    width: 0.25px;
```
Comment (mmhuntsberry) [inline src/components/MainContent/_maincontent.scss]: Should be 1px.  This is something you would probably talk to your design team about and tell them that there aren't quarter pixels.

File: src/components/MainContent/_maincontent.scss [hunk 6]
```diff
@@ -5,25 +5,35 @@ $gradient-background: linear-gradient(
   rgba(230, 214, 255, 1) 50%,
   rgba(243, 243, 243, 1) 100%
 );
+$background-extend-left: -10.3333%;
+$background-extend-right: -8.33333%;
 
 .section {
-  padding: 80px 0;
+  padding-top: 40px;
```
Comment (mmhuntsberry) [inline src/components/MainContent/_maincontent.scss:12]: Use spacing / layout token for padding and margins

File: src/components/MainContent/_maincontent.scss [hunk 7]
```diff
@@ -38,17 +48,26 @@ $gradient-background: linear-gradient(
...
-    background: $gradient-background;
-    z-index: -1;
+  @media screen and (min-width: 672px) {
+    flex-direction: row;
+  }
+
+  @media screen and (min-width: 1056px) {
+    margin-bottom: inherit;
+
+    &::before {
```
Comment (mmhuntsberry) [inline src/components/MainContent/_maincontent.scss:61]: You should make this &::before and after thing a `mixin` so you don't have to repeat code over and over again.

File: src/components/MainContent/_maincontent.scss [hunk 8]
```diff
@@ -85,6 +104,11 @@ $gradient-background: linear-gradient(
 }
 .card__description {
   @include carbon--type-style('body-short-01');
+  margin-bottom: 32px; // add spacing token
```
Comment (mmhuntsberry) [inline src/components/MainContent/_maincontent.scss:107]: listen to your comment lol

File: src/components/MainContent/_maincontent.scss [hunk 9]
```diff
@@ -111,13 +134,23 @@ $gradient-background: linear-gradient(
...
-  display: block;
+.wrappper--get-started,
+.wrapper--learn-more > .section__title,
+.section__title--subtle {
+  margin-bottom: $spacing-05;
+}
+
+.wrapper--get-started {
+  margin-right: calc(#{$layout-01} + #{$layout-07});
+  margin-bottom: 40px;
```
Comment (mmhuntsberry) [inline src/components/MainContent/_maincontent.scss:149]: spacing token

File: src/components/MainContent/MainContent.js [hunk 10]
```diff
@@ -10,8 +9,8 @@ const MainContent = (props) => {
       <section className="section section--signup bx--offset-lg-1 bx--col-lg-14">
         <h3 className="section__title">Why sign up?</h3>
         <div className="wrapper--card">
-          {props.details.map((detail) => {
-            return <Card detail={detail} />;
+          {props.details.map((detail, key) => {
+            return <Card detail={detail} key={key} />;
```
Comment (mmhuntsberry) [inline src/components/MainContent/MainContent.js:13]: You could use an implicit return here so you don't have to use the return keyword.

Comment (sydbails) [reply src/components/MainContent/MainContent.js:13]: Is this a small nit or something I should be doing moving forward?

**Comment by mmhuntsberry** [comment]
It's a small nit. :)

**Pull request merged**